Modules/IO/PhilipsREC/src/itkPhilipsRECImageIO.cxx

```diff
@@ -190,18 +190,16 @@ PhilipsRECImageIOGetImageTypeOffset(int
                                     PhilipsPAR::PARSliceIndexImageTypeVector    sliceImageTypesIndex,
                                     PhilipsPAR::PARSliceIndexScanSequenceVector sliceScanSequenceIndex)
 {
-  int index = volumeIndex * parParam.num_slice_repetitions * numSlices + slice * parParam.num_slice_repetitions;
-  int i;
-
-  for (i = 0; i < parParam.num_slice_repetitions; ++i)
+  const int index = volumeIndex * parParam.num_slice_repetitions * numSlices + slice * parParam.num_slice_repetitions;
+  for (int i = 0; i < parParam.num_slice_repetitions; ++i)
   {
     if ((sliceImageTypesIndex[index + i].second == imageType) &&
         (sliceScanSequenceIndex[index + i].second == scanSequence))
     {
-      break;
+      return i;
     }
   }
-  return i;
+  return 0; // Default to returning 0 if criteria is never met
```

Comment:
This is a behavior change, right? I guess it originally returned `parParam.num_slice_repetitions`, rather than `0`,  in that case. Did you test it?